lib/controllers/file-patch-controller.js

```diff
@@ -37,21 +37,24 @@ export default class FilePatchController extends React.Component {
     );
 
     this.state = {
-      lastFilePatch: this.props.filePatch,
       selectionMode: 'hunk',
       selectedRows: new Set(),
     };
 
     this.mouseSelectionInProgress = false;
     this.stagingOperationInProgress = false;
 
+    this.lastPatchString = null;
```

Comment:
What is the benefit of storing `lastPatchString` as a variable vs keeping it in the state?

Reply:
The state key was actually unused entirely. There were no references to it; the patch change promise was being resolved with a comparison between `this.props` and the `prevProps` given to `componentDidUpdate()`.

As for storing it as a string though - as part of the process of rendering a new patch, we "adopt" the TextBuffer from the previous one, which leaves the buffer on the old patch empty, so _every_ patch was being detected as "different". Pre-rendering the old one to a string lets us compare them even after we've messed around with it.